### PR TITLE
Better error message for 16 byte alignment error

### PR DESCRIPTION
**Pull request by IceSentry**
**Checklist**

- [x] Run `cargo clippy`.
- [x] Run `RUSTFLAGS=--cfg=web_sys_unstable_apis cargo clippy --target wasm32-unknown-unknown` if applicable.
- [x] Add change to CHANGELOG.md. See simple instructions inside file.

**Connections**
(Based on) Closes https://github.com/gfx-rs/wgpu/pull/3099
Closes https://github.com/gfx-rs/wgpu/issues/2832

**Description**
Previous error message was the generic downflags error message for BUFFER_BINDINGS_NOT_16_BYTE_ALIGNED that didn't highlight that if it was 16 bit aligned then this is supported. But it wasn't clear which shader group / bind was not aligned.

New error message:
```
In Device::create_render_pipeline
      note: label = `cuboids_pipeline`
    shader group 1 binding 1 must be 16 bytes aligned as device does not support DownlevelFlags `BUFFER_BINDINGS_NOT_16_BYTE_ALIGNED`
```

Also have switched the code around to check the flag once as if it's set then we can skip the check entirely.

**Testing**
I didn't test it myself, but the original author tested and I simply updated the message to address the comment on the PR

### CODE REVIEW

File: wgpu-core/src/pipeline.rs [hunk 1]
```diff
@@ -373,6 +373,8 @@ pub enum CreateRenderPipelineError {
         stage: wgt::ShaderStages,
         error: String,
     },
+    #[error("shader group {group} binding {binding} must be 16 bytes aligned as device does not support DownlevelFlags `BUFFER_BINDINGS_NOT_16_BYTE_ALIGNED`")]
```
Comment (cwfitzgerald) [inline wgpu-core/src/pipeline.rs:376]:
```suggestion
    #[error("In the provided shader, the type given for group {group} binding {binding} has an alignment of {alignment}. As the device does not support `DownlevelFlags::BUFFER_BINDINGS_NOT_16_BYTE_ALIGNED` the type must have an alignment of at least 16.")]
```

Sorry, one more round on this error message - note the extra argument alignment. Adding the eixsting alignment really helps make the message clear.

Comment (IceSentry) [reply wgpu-core/src/pipeline.rs]: @cwfitzgerald I'm trying to get the alignment, but I'm not sure how to go about this. Do I just get the size of the binding?

**Comment by IceSentry** [comment]
@cwfitzgerald btw, I replied to the code suggestion, but I know github notifications are a bit weird with resolved discussions so I'll ping you here.

**Comment by cwfitzgerald** [comment]
@IceSentry Good shout on the ping - I didn't see this. I just realized the requirement is that the _size_, not the alignment, is a multiple of 16. So my suggestion is actually slightly wrong. /s/alignment/size should take care of it. The requirement should be "a multiple of 16".

**Review by cwfitzgerald — APPROVED** [review]
Party party party

**Pull request merged**